File: aoc2204.py
a = []
# with open('test') as file:
with open('2204.0') as file:
    for l in file:
        a.append(l.strip())
res1 = 0
res2 = 0
for line in a:
    L, R = line.split(',')
    l, r = [int(_) for _ in L.split('-')]
    ll, rr = [int(_) for _ in R.split('-')]
    # print(l, r, ll, rr)
    if (l <= ll and r >= rr) or (ll <= l and rr >= r):
        # print(n)
        res1 += 1
    if r >= ll and l <= rr:
        res2 += 1

print('Star 1:', res1)
print('Star 2:', res2)

File: aoc2204.go
package main

import "fmt"
import "bufio"
import "os"
import _"strings" // strings.Contains(hays, "need")
import _"io/ioutil"
import _"strconv"

func solve(path string) (int, int) {
	data, err := os.Open(path)
	if err != nil {
		panic(err)
	}
	scanner := bufio.NewScanner(data)
	scanner.Split(bufio.ScanLines)
	res1, res2 := 0, 0
	for scanner.Scan() {
		var l, r, ll, rr int
		fmt.Sscanf(scanner.Text(), "%d-%d,%d-%d", &l, &r, &ll, &rr);
		// fmt.Println(l, r, ll, rr)
		if (l <= ll && r >= rr) || (ll <= l && rr >= r) {
			res1++
		}
		if r >= ll && l <= rr {
			res2++
		}
	}
	return res1, res2
}

func main() {
	var r1, r2 int

	r1, r2 = solve("_inputs/2204.0")
	fmt.Println("data")
	fmt.Println("Star 1:", r1)
	fmt.Println("Star 2:", r2, "\n")

	r1, r2 = solve("_inputs/2204.1")
	fmt.Println("test")
	fmt.Println("Star 1:", r1)
	fmt.Println("Star 2:", r2)
}
